builtin/plugin/backend.go

```diff
@@ -27,13 +29,76 @@ func Factory(conf *logical.BackendConfig) (logical.Backend, error) {
 // Backend returns an instance of the backend, either as a plugin if external
 // or as a concrete implementation if builtin, casted as logical.Backend.
 func Backend(conf *logical.BackendConfig) (logical.Backend, error) {
+	var b backend
 	name := conf.Config["plugin_name"]
 	sys := conf.System
 
-	b, err := bplugin.NewBackend(name, sys, conf.Logger)
+	raw, err := bplugin.NewBackend(name, sys, conf.Logger)
 	if err != nil {
 		return nil, err
 	}
+	b.Backend = raw
+	b.config = conf
 
-	return b, nil
+	return &b, nil
+}
+
+// backend is a thin wrapper around plugin.BackendPluginClient
+type backend struct {
+	logical.Backend
+	sync.Mutex
```

Comment:
That's a good thing! Otherwise every request that comes into the backend could cause a reload to happen